lib/node.dart

```diff
@@ -194,6 +240,15 @@ base class Node implements SceneGraph {
     }
   }
 
+  /// Prepares and renders this node and its children within the scene graph.
+  ///
+  /// This method calculates the final world transform for the node by combining the
+  /// parent's world transform with the node's local transform. It then renders the node's
+  /// mesh primitives using the provided `SceneEncoder`, and recursively renders all child nodes.
+  ///
+  /// The rendering process involves encoding the node's geometry and material data into commands
+  /// that are passed to the GPU for drawing. If the node has associated skinning data (for animated models),
+  /// this data is also passed along to the rendering pipeline.
```

Comment:
```suggestion
  /// Recursively records mesh draw operations for this node and all its children.
  ///
  /// To display this node in a dart:ui canvas, add this node to a `Scene` and call `Scene.render` instead.
```

Perhaps we should keep the docstring for this one simple. Users won't normally call this directly. We can guide them to using `Scene.render` instead.